providers/teams/pages/add.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { ChangeEvent } from 'react'
+import dynamic from 'next/dynamic'
+import { gql } from 'apollo-boost'
+import { useRouter } from 'next/router'
+import { apolloClient } from '../lib/apollo'
+import cookie from 'js-cookie'
+import { decode } from 'jsonwebtoken'
+import { Team } from '../lib/common'
+
+/**
+ * Note: we're sending the variables to grahpql by the input's name.
+ */
+export const AddTeamPage = props => {
+    const router = useRouter()
+    const userId = getUserId()
+
+    async function onSubmit(event: ChangeEvent<HTMLFormElement>) {
+        event.preventDefault()
+        const formEle = document.forms[0]
+        const formData = new FormData(formEle)
+        const team: Team = {
+            name: formData.get('name').toString(),
+            description: formData.get('description').toString(),
+        }
+        // TODO do we need to check whether he's already on a team, and block if he is?
```

Comment:
Yes

Reply:
couldnt we do that at the Hasura level?

Reply:
Do you mean on a model level? How would you like to do it?

Reply:
Also, I'm missing where this is resolved, please help me (: